pkg/gatewayserver/io/io.go

```diff
@@ -274,6 +274,12 @@ func (c *Connection) HandleUp(up *ttnpb.UplinkMessage, frontendSync *FrontendClo
 	}
 
 	for _, md := range up.RxMetadata {
+		r := receivedAt
+		if _, _, median, _, count := c.RTTStats(100, time.Now()); count > 0 {
```

Comment:
Two things:
1. We want the message's received time as reference time. The difference between `time.Now()` and `receivedAt` is probably negligible, but still, it's cleaner to use `receivedAt` here I think
2. The result of the `c.RTTStats()` does not depend on the loop variable `md`, so this calculation should be moved to before the loop, and the loop is only needed if there's a known RTT to apply